JetNews/app/src/main/java/com/example/jetnews/ui/interests/InterestsScreen.kt

```diff
@@ -287,13 +290,14 @@ private fun TabWithTopics(
  */
 @Composable
 private fun TabWithSections(
+    windowSize: WindowSize,
     sections: List<InterestSection>,
     selectedTopics: Set<TopicSelection>,
     onTopicSelect: (TopicSelection) -> Unit
 ) {
+    val columns = remember(windowSize) { if (windowSize == WindowSize.Compact) 1 else 2 }
```

Comment:
Just wanted to highlight this case:
We are keeping the `BoxWithConstraints`, because we want to base some decision around the local space available.
However, we are also making a decision based on the window size class, which we pass in.

Reply:
SG. In the two cases where we're using BWC at the moment, should we add a TODO saying: "We could optimize this code by removing BoxWithConstraints and use a custom Layout that places elements on the screen given the available width"?

Reply:
Added!